### ISSUE TITLE
Remove Notebooks, keeping their test and doc info

### ISSUE THREAD

**Issue by alexanderzimmerman** [cgre-aachen/gempy#273]
There are two big items here:
- Move all tests in Jupyter Notebooks to .py files
- Use Sphinx-Gallery to run examples and embed outputs into readthedocs website

These must go to a single PR. The problem is that removing the notebooks also removes a lot of documentation.

## Tests overhaul
* [x] Convert all Jupyter notebooks to Python scripts (.py)
* [x] Run tests with pytest, locally and with Travis. For now, it is only important that all of the lines of code from the notebooks are still run successfully. No need for new asserts.
* [x] Bonus points: Replace redundant model creation with test fixtures ( #319 )

## Embed script outputs into docs
All documentation that was previously captured in Jupyter Notebooks should now be embedded in readthedocs site using Sphinx-Gallery, e.g. https://docs.pyvista.org/examples/index.html
* [x] Try Sphinx-Gallery for embedding outputs into HTML ( #292, #308 )
* [x] Integrate the new Sphinx-Gallery content with our automated docs publishing. readthedocs should have a Sphinx-Gallery plugin (but we haven't found it yet). Alternatively: Run sphinx locally (with Sphinx-Gallery plugin) and make a GitHub action ( #293 ) to push docs to GitHub web page.

**Comment by alexanderzimmerman** [comment]
This issue is high on our priority list for GemPy 2.2. @WestfalNamur , could you edit the original description above so that the goals and requirements are more clear? Right now the description still has the default text from the issue template.

**Comment by alexanderzimmerman** [comment]
@Leguark , does "all notebooks" really mean "all notebooks" in the repo or are there some that should not be converted to .py?

**Comment by alexanderzimmerman** [comment]
Also before I work on this, is [the project board](https://github.com/cgre-aachen/gempy/projects/24) up to date? Is it relevant or is it superseded by the checklist in this PR's description?

**Comment by alexanderzimmerman** [comment]
This apparently wasn't linked to #308 yet.

**Comment by alexanderzimmerman** [comment]
Also, @Leguark , do I understand correctly that part of this Epic is to no longer use any Notebooks as tests? Before that was a separate issue from Sphinx in my head.

**Comment by alexanderzimmerman** [comment]
So in principle we could have an issue to remove Notebooks as tests, replacing them with .py files. The problem is that then we also lose a lot of documentation. Therefore, this Epic tackles both the test overhaul and also the Sphinx docs overhaul using Sphinx Gallery.

**Comment by alexanderzimmerman** [comment]
I'm going to rewrite the description to be inline with my understanding of the issue.

**Comment by Leguark** [comment]
yes, we need to have the sphinx gallery running before we remove the notebooks but configuring pytest to run in gempy/examples should be quite straightforward

**Comment by alexanderzimmerman** [comment]
I am waiting on #380 and for this issue to be updated with how to proceed. Please re-assign me when I can start working on it.
